packages/frontend/editor-ui/src/components/Node/NodeCreator/utils.ts

```diff
@@ -90,22 +90,30 @@ export function sortNodeCreateElements(nodes: INodeCreateElement[]) {
 	});
 }
 
+// We remove `Trigger` from e.g. `Telegram Trigger` to show it as part of the `Telegram` group,
+// but still want to show matching results when the user types `Telegram Tri` or `Telegram Trigger`
+// Ideally this would be handled via metadata, but that is a larger refactor.
+export function removeTrailingTrigger(searchFilter: string) {
+	const parts = searchFilter.split(' ');
+	if (parts.length > 1 && 'trigger'.startsWith(parts.slice(-1)[0]?.toLowerCase() ?? 'a')) {
+		return parts.slice(0, -1).join(' ').trimEnd();
+	}
+	return searchFilter;
```

Comment:
[nits] For more robust whitespace handling and clarity (if you don't hate regex)
```suggestion
  const [, prefix = ''] = searchFilter.match(/^(.*)\s+trigger$/i) ?? [];
    
  return prefix.trim();
```

Reply:
Tbh I spent like 5 minutes just now between this and the docs for match and I still don't fully understand what I'm getting back from the match call.
Based on that I'd stick with my implementation which I feel is more explicit about when we don't change the input at all